File: int_overflow.c
#include <stdio.h>
#include <stdint.h>

int main() {
	uint8_t u8_val = 0;
	int8_t i8_val = 0;
	int8_t input;

	while (1) {
		printf("Enter i8 values: ");
		scanf("%hhd", &input);

		u8_val += input;
		i8_val += input;
	
		printf("Input: %d, u8: %u, i8: %d\n", input, u8_val, i8_val);
	}
	return 0;
}

File: int_overflow.rs
use std::io;

fn main() {
	let mut u8_val: u8 = 0;
	let mut i8_val: i8 = 0;
	
	loop {
		println!("Enter i8 value: ");
		let mut input = String::new();
		io::stdin().read_line(&mut input).unwrap();
		let num: i8 = input.trim().parse().unwrap();

		u8_val += num as u8;
		i8_val += num;
	
		println!("Input: {}, u8: {}, i8: {}", num, u8_val, i8_val);
	}
}
